src/net.cpp

```diff
@@ -1451,8 +1447,10 @@ void CConnman::SocketEvents(std::set<SOCKET> &recv_set, std::set<SOCKET> &send_s
 
 void CConnman::SocketHandler()
 {
+    NodesSnapshot snap{*this};
+
     std::set<SOCKET> recv_set, send_set, error_set;
-    SocketEvents(recv_set, send_set, error_set);
+    SocketEvents(snap.Nodes(), recv_set, send_set, error_set);
```

Comment:
Could we move this call to `SocketEvents()` to below the _Accept new connections_ logic? I think it's simpler to reason about if we're not holding the nodes snapshot across the _Accept new connections_ logic.

Reply:
Sorry, ignore this. I've just realized that the _Accept new connections_ logic is using the `recv_set` that's populated by `SocketEvents()`.